File: src/main/java/com/lumiyaviewer/lumiya/render/avatar/AnimationTiming.java
package com.lumiyaviewer.lumiya.render.avatar;

public class AnimationTiming {
    public float inAnimationTime = 0.0f;
    public float inFactor = 0.0f;
    public float outFactor = 1.0f;
    public float runningTime = 0.0f;

    public boolean hasStopped() {
        return this.outFactor <= 0.0f;
    }
}

File: Linkpoint/src/main/app_backup/com/lumiyaviewer/lumiya/render/avatar/AnimationTiming.kt
package com.lumiyaviewer.lumiya.render.avatar

open class AnimationTiming {
    var inAnimationTime = 0.0f
    var inFactor = 0.0f
    var outFactor = 1.0f
    var runningTime = 0.0f

    fun hasStopped(): Boolean = outFactor <= 0.0f
}
